assets/components/productPage/productPagePeriodForm/productPagePeriodForm.jsx

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React from 'react';
+
+import SvgInfo from 'components/svgs/information';
+import { outsetClassName, bgClassName } from 'components/productPage/productPageContentBlock/productPageContentBlock';
+
+import ProductPageButton from '../productPageButton/productPageButton';
+import ProductPagePeriodFormLabel from './productPagePeriodFormLabel';
+
+// ---- Types ----- //
+
+export type Period = {|
+  title: string,
+  copy: string,
+  offer: string | null,
+|}
+
+export type StatePropTypes<P> = {|
+  periods: {[P]: Period},
+  selectedPeriod: P | null,
+|};
+
+export type DispatchPropTypes<P> = {|
+  onSubmitAction: () => *,
+  setPeriodAction: (P) => *,
+|};
+
+type PropTypes<P> = {|
+  ...StatePropTypes<P>,
+  ...DispatchPropTypes<P>,
+|};
+
+
+// ----- Render ----- //
+
+export default function ProductPagePeriodForm<P:string>({
+  periods, selectedPeriod, onSubmitAction, setPeriodAction,
+}: PropTypes<P>) {
+
+  const keys = Object.keys(periods);
+
+  return (
+    <form
+      className="component-product-page-period-form-wrap"
```

Comment:
The top-level `className` takes the name of the component, in this case `component-product-page-period-form`.